WorkloadManagementSystem/DB/JobDB.py

```diff
@@ -566,20 +553,18 @@ def setNextOptimizer(self, jobID, currentOptimizer):
     if not result['OK']:
       return result
 
-    optListString = result['Value']
-    optList = optListString.split(',')
+    optList = result['Value'].split(',')
+    if currentOptimizer not in optList:
+      return S_ERROR('Could not find ' + currentOptimizer + ' in chain')
     try:
-      sindex = None
-      for i in xrange(len(optList)):
-        if optList[i] == currentOptimizer:
-          sindex = i
-      if sindex >= 0:
-        if sindex < len(optList) - 1:
-          nextOptimizer = optList[sindex + 1]
-        else:
-          return S_ERROR('Unexpected end of the Optimizer Chain')
-      else:
-        return S_ERROR('Could not find ' + currentOptimizer + ' in chain')
+      # Append None to get a list of (opt,nextOpt)
+      optList.append(None)
+      nextOptimizer = None
+      for opt, nextOptimizer in zip(optList[:-1], optList[1:]):
```

Comment:
`zip` changes in python3: instead of a `list` it would return an iterator. There's a fix explained in https://portingguide.readthedocs.io/en/latest/iterators.html#new-behavior-of-zip which should be used here.

Reply:
I am not sure I understand, why should it be used here ? There is no modification of optList, is there ?

Reply:
I tend to agree with Chris... Is there a real problem there?

Reply:
I approved this PR because as you say there's no specific issue with these lines of code, but I left this comment because we should in general pay attention to use `zip` (and not only) from now on. Sorry, I realize I wasn't very clear.

Reply:
I think it is more the other way around, just like for `dict.items()`: we always use the iterator version (`iteritem`) except when we really need a copy. And I think that this is what the automated migration tools expect. So in that case, I think we should specifically do `list(zip())` when we need a copy, even if in py2, it will do a double copy...
But yes, we should be damn careful with that

Reply:
... and check all the `zip()` occurrences in the current code! I think this is a mistake from python to change the behaviour like that... they could have introduced a `zipiter()` or `zipitems()` method like for the `dict` methods...
Thanks anyway for the heads up!